torch/optim/_functional.py

```diff
@@ -322,3 +322,43 @@ def adamax(params: List[Tensor],
         clr = lr / bias_correction
 
         param.addcdiv_(exp_avg, exp_inf, value=-clr)
+
+
+def asgd(params: List[Tensor],
+         grads: List[Tensor],
+         axs: List[Tensor],
+         mus: List[float],
+         etas: List[float],
+         state_steps: List[int],
```

Comment:
nit: do we need steps here?

Reply:
thanks for pointing out, indeed stete_steps, lr, alpha and t0 are unnecessary, but i believe weight_decay and lambd are used inside the function.